src/HealthGPS/static_linear_model.cpp

```diff
@@ -0,0 +1,118 @@
+#include "static_linear_model.h"
+#include "runtime_context.h"
+
+#include "HealthGPS.Core/exception.h"
+
+namespace hgps {
+
+StaticLinearModel::StaticLinearModel(std::vector<core::Identifier> risk_factor_names,
+                                     LinearModelParams risk_factor_models,
+                                     Eigen::MatrixXd risk_factor_cholesky)
+    : risk_factor_names_{std::move(risk_factor_names)},
+      risk_factor_models_{std::move(risk_factor_models)},
+      risk_factor_cholesky_{std::move(risk_factor_cholesky)} {
+
+    if (risk_factor_names_.empty()) {
+        throw core::HgpsException("Risk factor names list is empty");
+    }
+    if (risk_factor_models_.intercepts.empty() || risk_factor_models_.coefficients.empty()) {
+        throw core::HgpsException("Risk factor models mapping is incomplete");
+    }
+    if (!risk_factor_cholesky_.allFinite()) {
+        throw core::HgpsException("Risk factor Cholesky matrix contains non-finite values");
+    }
+}
+
+HierarchicalModelType StaticLinearModel::type() const noexcept {
+    return HierarchicalModelType::Static;
+}
+
+std::string StaticLinearModel::name() const noexcept { return "Static"; }
+
+void StaticLinearModel::generate_risk_factors(RuntimeContext &context) {
+
+    for (auto &person : context.population()) {
+
+        // Approximate risk factor values with linear models.
+        linear_approximation(person);
+
+        // Correlated residual sampling.
+        auto samples = correlated_samples(context);
+    }
+}
+
+void StaticLinearModel::update_risk_factors(RuntimeContext &context) {
```

Comment:
Not in this case, as `update_risk_factors` is an override, where other models, including this one eventually, will need write access to the context.

Reply:
... actually, that might not be true. Right now they just need write access to the `Population` contained in context. I'll open an issue for later, since it requires changing the RF model interface.